std/traits.d

```diff
@@ -5563,15 +5563,38 @@ enum bool isSomeChar(T) = is(CharTypeOf!T) && !isAggregateType!T;
     static assert(!isSomeChar!(S!char));
 }
 
-/**
-Detect whether $(D T) is one of the built-in string types.
-
-The built-in string types are $(D Char[]), where $(D Char) is any of $(D char),
-$(D wchar) or $(D dchar), with or without qualifiers.
-
-Static arrays of characters (like $(D char[80])) are not considered
-built-in string types.
- */
+/++
+    Detect whether `T` is one of the built-in string types.
+
+    A built-in string type is any array type whose element type is `char`,
+    `wchar`, or `dchar` - regardless of qualifiers (that includes `string`,
+    `wstring`, and `dstring`).
+
+    All arrays whose element types are `char`, `wchar`, or `dchar` -
+    regardless of qualifiers (that includes `string` and `wstring`). It
+    specifically does $(I not) include arrays of `dchar`.
+
+    Static arrays of characters (like `char[80]`) are not considered built-in
+    string types.
+
+    $(RED Warning:
+        This trait is true for enums with a base type that is a built-in string
+        type. In some cases, this is useful, but in many cases, for a function
+        to work correctly with both an enum and the enum's base type, the enum
+        value needs to be converted to the base type (which won't happen
+        automatically if the function is templated on the parameter type rather
+        than expicitly accepting the enum's base type); otherwise, it risks not
```

Comment:
This is rather wordy. Perhaps split the sentence in the parentheses into a new sentence? And begin a new sentence on the "otherwise".